features/issues/api/use-get-issues.tsx

```diff
@@ -1,33 +1,72 @@
-import { useEffect } from "react";
-import { useQuery, useQueryClient } from "@tanstack/react-query";
-import { getIssues } from "@api/issues";
+import { useQuery } from "@tanstack/react-query";
+import { getAllIssues, getIssues } from "@api/issues";
 import type { Page } from "@typings/page.types";
 import type { Issue } from "@api/issues.types";
 
-const QUERY_KEY = "issues";
+export const statusMapping: Record<string, string> = {
+  resolved: "resolved", // User-facing "Resolved" maps to API "resolved"
+  unresolved: "open", // User-facing "Unresolved" maps to API "open"
+};
 
-export function getQueryKey(page?: number) {
-  if (page === undefined) {
-    return [QUERY_KEY];
-  }
-  return [QUERY_KEY, page];
-}
+export const reverseStatusMapping: Record<string, string> = {
+  resolved: "resolved", // API "resolved" maps back to "Resolved"
+  open: "unresolved", // API "open" maps back to "Unresolved"
+};
+
+export function useGetIssues(
+  page: number,
+  searchTerm?: string,
+  status?: string,
+  level?: string,
+) {
+  // console.log("useGetIssues - level:", level);
+  const queryKey =
+    searchTerm || status || level
+      ? ["issues", "all", searchTerm, status, level]
+      : ["issues", page];
+  // console.log("useGetIssues - queryKey:", queryKey);
+  const fetchData =
+    searchTerm || status || level
+      ? () => getAllIssues() // Fetch all items for filtering
+      : (options: { signal?: AbortSignal }) => getIssues(page, options); // Paginated fetch for default view
```

Comment:
This conditional fetching isn't necessary actually. The API endpoint already accepts the filter params and does the filtering and pagination. So the `getAllIssues` function is in fact redundant. 

No worries, I've seen quite a few people taken this approach with local filtering. But with paginated endpoints the filtering always has to be done on the server. Otherwise you'd have overcomplicated code as you see here.